### PR TITLE
CashFusion: more flexible fee fuzzing logic; more funds safety checks

### PR DESCRIPTION
**Pull request by markblundeberg**
We want to make the 'fee fuzzing' get larger for the higher tiers, so it (and the 'excess fee') needs to be tier-dependent instead of fixed.

At the same time, add more funds/fees sanity checks.

(Have tested this change with dozens of fusions, it is working smoothly.)

### CODE REVIEW

File: plugins/fusion/fusion.py [hunk 1]
```diff
@@ -364,7 +373,8 @@ def check_coins(self):
     def clear_coins(self):
         """ Clear the inputs list and release frozen coins. """
         for wallet, (coins, mytxids, checked_txids) in self.source_wallet_info.items():
-            wallet.set_frozen_coin_state(coins, False)
+            with wallet.lock:
```
Comment (cculianu) [inline plugins/fusion/fusion.py:376]: Why lock here? ` set_frozen_coin_state` already takes this lock when it is performing its operations.

Comment (markblundeberg) [reply plugins/fusion/fusion.py:376]: d'oh, you're right -- I was looking at the wrong one (`set_frozen_state`)!

Comment (cculianu) [reply plugins/fusion/fusion.py:376]: Ok.. waiting for your unfix to the non-fixing-anything fix.

**Comment by cculianu** [comment]
OK, well I don't see any major foot-guns here.  As far as the details of how cashfusion works, I defer to your expertise there and trust you "did the right thing".  Since you tested it I would say -- I trust it works.

You said Jonald wanted to also look at this so I will hold off merging and recommend you or Jonald merge this when you are satisfied.

**Pull request merged**